src/System.IO.Pipes/src/System/IO/Pipes/PipeAsyncResult.cs

```diff
@@ -9,12 +9,11 @@ namespace System.IO.Pipes
 {
     internal unsafe sealed class PipeAsyncResult : IAsyncResult
     {
+        internal NamedPipeServerStream _stream;
```

Comment:
Why not just make this a ThreadPoolBoundHandle?  Do you need to root the NamedPipeServerStream?